src/snowflake/cli/api/project/schemas/entities/common.py

```diff
@@ -59,6 +61,25 @@ def get_type(cls) -> str:
         return cls.model_fields["type"].annotation.__args__[0]
 
     meta: Optional[MetaField] = Field(title="Meta fields", default=None)
+    identifier: Optional[Union[Identifier | str]] = Field(
+        title="Entity identifier", default=None
+    )
+    key: str | None = None
+
+    @model_validator(mode="after")
+    def validate_identifier(self) -> EntityBase:
+        if not self.key and not self.identifier:
+            raise ValueError("Missing entity identifier")
+        return self
+
+    @property
+    def fqn(self) -> FQN:
```

Comment:
Are we sure each entity will have a FQN?

Reply:
Very good question. My thinking for now is:
1. Usage of `fqn` is not automated. When developing code around entity developers need to determine if they are working with "fqn" entity or not.
2. Using fqn for non-snowflake entities will probably work in majority of cases (as long as it complies with regexp). 
3. Once we start having non-snowflake entities we may consider differentiating them, `SnowflakeEntity` vs `Entity`.

WDYT?

Reply:
Sounds good. Let's re-discuss once we have our first non-snowflake entity.